subprojects/core-api/src/main/java/org/gradle/api/DomainObjectCollection.java

```diff
@@ -136,6 +151,9 @@ public interface DomainObjectCollection<T> extends Collection<T> {
     /**
      * Adds a closure to be called when an object is removed from this collection. The removed object is passed to the
      * closure as the parameter.
+     * <p>
+     * This method will only be invoked for elements effectively added to the collection, and might cause their realization.
```

Comment:
Same comment about "might" and "observed" as above.

Reply:
Same answers 😉